[BUG] Running integration tests with `PERFILE` results in failed tests

**Describe the bug**
Running some tests that limit the ColumnSizeBytes results in failures when running with parquet reader type `PERFILE`

**Steps/Code to reproduce bug**
```
SPARK_HOME=~/dist/spark-3.5.5-bin-hadoop3-scala2.13 PYSP_TEST_spark_rapids_sql_format_parquet_reader_type=PERFILE ./integration_tests/run_pyspark_from_build.sh  -k 'test_parquet_partition_batch_row_count_only_splitting or test_col_size_exceeding_cudf_limit'

============================================================================================== short test summary info ==============================================================================================
FAILED ../../../../integration_tests/src/main/python/col_size_exceeding_cudf_limit_test.py::test_col_size_exceeding_cudf_limit[single_partition_single_value_without_nulls-parquet][DATAGEN_SEED=1743641247, TZ=UTC, INJECT_OOM] - AssertionError: CPU and GPU list have different lengths at [] CPU: 5000 GPU: 10
FAILED ../../../../integration_tests/src/main/python/col_size_exceeding_cudf_limit_test.py::test_col_size_exceeding_cudf_limit[single_partition_multiple_value-parquet][DATAGEN_SEED=1743641247, TZ=UTC] - AssertionError: CPU and GPU list have different lengths at [] CPU: 5000 GPU: 1694
FAILED ../../../../integration_tests/src/main/python/col_size_exceeding_cudf_limit_test.py::test_col_size_exceeding_cudf_limit[single_partition_single_empty_value_with_nulls-parquet][DATAGEN_SEED=1743641247, TZ=UTC, INJECT_OOM] - AssertionError: CPU and GPU list have different lengths at [] CPU: 5000 GPU: 3373
FAILED ../../../../integration_tests/src/main/python/col_size_exceeding_cudf_limit_test.py::test_col_size_exceeding_cudf_limit[multiple_partition_single_value-parquet][DATAGEN_SEED=1743641247, TZ=UTC] - AssertionError: CPU and GPU list have different lengths at [] CPU: 5000 GPU: 1275
FAILED ../../../../integration_tests/src/main/python/col_size_exceeding_cudf_limit_test.py::test_col_size_exceeding_cudf_limit[multiple_partition_multiple_value_wider_first_col-parquet][DATAGEN_SEED=1743641247, TZ=UTC, INJECT_OOM] - AssertionError: CPU and GPU list have different lengths at [] CPU: 5000 GPU: 714
FAILED ../../../../integration_tests/src/main/python/col_size_exceeding_cudf_limit_test.py::test_col_size_exceeding_cudf_limit[multiple_partition_int_value-parquet][DATAGEN_SEED=1743641247, TZ=UTC] - AssertionError: CPU and GPU list have different lengths at [] CPU: 5000 GPU: 2515
FAILED ../../../../integration_tests/src/main/python/col_size_exceeding_cudf_limit_test.py::test_col_size_exceeding_cudf_limit[multiple_partition_multiple_value_narrow_first_col-parquet][DATAGEN_SEED=1743641247, TZ=UTC, INJECT_OOM] - AssertionError: CPU and GPU list have different lengths at [] CPU: 5000 GPU: 714
FAILED ../../../../integration_tests/src/main/python/parquet_test.py::test_parquet_partition_batch_row_count_only_splitting[DATAGEN_SEED=1743641247, TZ=UTC, INJECT_OOM] - AssertionError: CPU and GPU list have different lengths at [] CPU: 1000 GPU: 100

```

**Expected behavior**
The above tests should pass or throw an exception not result in data corruption

**Additional Context**
https://github.com/NVIDIA/spark-rapids/pull/12396#issuecomment-2756870327
https://github.com/NVIDIA/spark-rapids/pull/12397#issuecomment-2774306783

--- a/integration_tests/src/main/python/col_size_exceeding_cudf_limit_test.py
+++ b/integration_tests/src/main/python/col_size_exceeding_cudf_limit_test.py
@@ -1,4 +1,4 @@
-# Copyright (c) 2023, NVIDIA CORPORATION.
+# Copyright (c) 2023-2025, NVIDIA CORPORATION.
 #
 # Licensed under the Apache License, Version 2.0 (the "License");
 # you may not use this file except in compliance with the License.
@@ -84,11 +84,15 @@ def extract_partition_cols(gen_list):
 @inject_oom
 @pytest.mark.parametrize('file_format', file_formats)
 @pytest.mark.parametrize('key', gen_list_dict.keys())
-def test_col_size_exceeding_cudf_limit(spark_tmp_path, file_format, key):
+@pytest.mark.parametrize('reader_type', ['AUTO', 'COALESCING', 'MULTITHREADED', 'PERFILE'])
+def test_col_size_exceeding_cudf_limit(spark_tmp_path, file_format, key, reader_type):
     gen_list = gen_list_dict[key]
     partition_cols = extract_partition_cols(gen_list)
     gen = StructGen(gen_list, nullable=False)
     data_path = spark_tmp_path + '/PART_DATA/' + key
+    new_conf=copy_and_update(conf, {'spark.rapids.sql.format.parquet.reader.type': reader_type,
+        'spark.rapids.sql.format.orc.reader.type': reader_type,
+        'spark.rapids.sql.format.avro.reader.type': reader_type})
     with_cpu_session(
         lambda spark: gen_df(spark, gen, length=5000).coalesce(1).write.partitionBy(partition_cols).format(file_format)
         .save(data_path))

--- a/sql-plugin/src/main/scala/com/nvidia/spark/rapids/ColumnarPartitionReaderWithPartitionValues.scala
+++ b/sql-plugin/src/main/scala/com/nvidia/spark/rapids/ColumnarPartitionReaderWithPartitionValues.scala
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2019-2023, NVIDIA CORPORATION.
+ * Copyright (c) 2019-2025, NVIDIA CORPORATION.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -33,9 +33,10 @@ class ColumnarPartitionReaderWithPartitionValues(
     partitionValues: InternalRow,
     partitionSchema: StructType,
     maxGpuColumnSizeBytes: Long) extends PartitionReader[ColumnarBatch] {
-  override def next(): Boolean = fileReader.next()
+  override def next(): Boolean = {
+    outputIter.hasNext || fileReader.next()
+  }
   private var outputIter: Iterator[ColumnarBatch] = Iterator.empty
-
 
   override def get(): ColumnarBatch = {
     if (partitionSchema.isEmpty) {

